### PR TITLE
Allow eager adjoint with qjit

### PR DESCRIPTION
**Pull request by dime10**
This PR just removes the restriction from the PennyLane side that in qjit mode `lazy` cannot be set to `False`. If a specific compiler does not support the flag it can always raise an error in its own implementation.

[sc-64178]

### CODE REVIEW

File: pennylane/ops/op_math/adjoint.py [hunk 1]
```diff
@@ -160,11 +159,9 @@ def loop_fn(i):
 
     """
     if active_jit := compiler.active_compiler():
-        if lazy is False:
-            raise CompileError("Setting lazy=False is not supported with qjit.")
         available_eps = compiler.AvailableCompilers.names_entrypoints
         ops_loader = available_eps[active_jit]["ops"].load()
-        return ops_loader.adjoint(fn)
+        return ops_loader.adjoint(fn, laze=lazy)
```
Comment (mlxd) [inline pennylane/ops/op_math/adjoint.py:164]: Wait, should this be `laze` or `lazy` ? I cannot see a `laze` option anywhere

Comment (dime10) [reply pennylane/ops/op_math/adjoint.py:164]:
```suggestion
        return ops_loader.adjoint(fn, lazy=lazy)
```

**Review by maliasadi — APPROVED** [review]
You also need to update integration tests,
```
FAILED tests/test_compiler.py::TestCatalyst::test_qjit_adjoint_lazy - TypeError: adjoint() got an unexpected keyword argument 'lazy'
FAILED tests/test_compiler.py::TestCatalyst::test_qjit_adjoint - TypeError: adjoint() got an unexpected keyword argument 'lazy'
```
Looks good to me otherwise :raised_hands:

**Comment by dime10** [comment]
> You also need to update integration tests,
> 
> ```
> FAILED tests/test_compiler.py::TestCatalyst::test_qjit_adjoint_lazy - TypeError: adjoint() got an unexpected keyword argument 'lazy'
> FAILED tests/test_compiler.py::TestCatalyst::test_qjit_adjoint - TypeError: adjoint() got an unexpected keyword argument 'lazy'
> ```
> 
> Looks good to me otherwise 🙌

Problem is these tests will fail until the Catalyst PR is merged, but the Catalyst PR relies on this PR. In the meantime I've updated the test and marked it xfail.

**Comment by mlxd** [comment]
> > You also need to update integration tests,
> > ```
> > FAILED tests/test_compiler.py::TestCatalyst::test_qjit_adjoint_lazy - TypeError: adjoint() got an unexpected keyword argument 'lazy'
> > FAILED tests/test_compiler.py::TestCatalyst::test_qjit_adjoint - TypeError: adjoint() got an unexpected keyword argument 'lazy'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Looks good to me otherwise 🙌
> 
> Problem is these tests will fail until the Catalyst PR is merged, but the Catalyst PR relies on this PR. In the meantime I've updated the test and marked it xfail.

So this PR goes green and goes in, then the catalyst PR goes in, and then this xfail test will need another update in this repo?

**Comment by dime10** [comment]
> > > You also need to update integration tests,
> > > ```
> > > FAILED tests/test_compiler.py::TestCatalyst::test_qjit_adjoint_lazy - TypeError: adjoint() got an unexpected keyword argument 'lazy'
> > > FAILED tests/test_compiler.py::TestCatalyst::test_qjit_adjoint - TypeError: adjoint() got an unexpected keyword argument 'lazy'
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Looks good to me otherwise 🙌
> > 
> > 
> > Problem is these tests will fail until the Catalyst PR is merged, but the Catalyst PR relies on this PR. In the meantime I've updated the test and marked it xfail.
> 
> So this PR goes green and goes in, then the catalyst PR goes in, and then this xfail test will need another update in this repo?

Actually I just realized that the Catalyst PR can be merged without this one. Only problem is I think PennyLane can only run tests against stable Catalyst 🤔

**Review by mlxd — APPROVED** [review]

**Pull request merged**